packages/fileeditor-extension/src/index.ts

```diff
@@ -174,6 +189,95 @@ function activate(app: JupyterLab, registry: IDocumentRegistry, restorer: ILayou
     return tracker.currentWidget !== null;
   }
 
+
+  class CodeSnippet {
+    startLine: number;
+    endLine: number;
+    code: string;
+    constructor(startLine: number) {
+      this.startLine = startLine;
+      this.code = "";
+    }
+  }
+
+  function findNextMarker(text: string) {
+    for (let marker of markdownMarkers) {
+      const index = text.indexOf(marker);
+      if (index > -1) {
+        return marker;
+      }
+    }
+    return '';
+  }
+  /**
+   * Construct all code snippets from current text
+   * (this could be potentially optimized if we can cache and detect differences)
+   */
+  function findCodeSnippets(): CodeSnippet[] {
```

Comment:
I think we should have a standalone function that takes a markdown string and returns a list of code block objects. I think there are other places we might want to use this function.